pkg/workflows/sdk/workflow_spec.go

```diff
@@ -35,3 +152,57 @@ func (w *WorkflowSpec) Steps() []StepDefinition {
 	s = append(s, w.Targets...)
 	return s
 }
+
+func (w *WorkflowSpec) FormatChart() (string, error) {
```

Comment:
Are we duplicating logic here? Why don't we just parse the graph and execute the template from that? A lot of the logic we've added here will reimplement the parsing logic.

Reply:
Probably :shrug: I wasn't looking outside of this package. Which type is best suited for this?

Reply:
Moved to a method on `DependencyGraph` but needs more refactoring to use the graph instead of custom code.